android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java

```diff
@@ -168,12 +169,10 @@ public void addTabItemInController(TiViewProxy tabProxy)
 		int index = this.mMenuItemsArray.size() - 1;
 		updateDrawablesAfterNewItem(index);
 		// Handle shift mode.
-		if (this.proxy.hasPropertyAndNotNull(TiC.PROPERTY_SHIFT_MODE)) {
-			if (!((Boolean) proxy.getProperty(TiC.PROPERTY_SHIFT_MODE))) {
-				this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_LABELED);
-			} else {
-				this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_SELECTED);
-			}
+		if (proxy.getProperties().optBoolean(TiC.PROPERTY_SHIFT_MODE, true)) {
+			this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_AUTO);
+		} else {
+			this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_UNLABELED);
```

Comment:
When "shiftMode" is set `false`, we're supposed to show titles for all tabs. Even if the tab is not selected. That was the old behavior. Most devs are setting it to `false` with the intention of showing titles. So, we should change it back to `LABEL_VISIBILITY_SELECTED`.

That said, if all tabs are not assigned a "title", then perhaps we should ignore the "shiftMode" property and set it `LABEL_VISIBILITY_UNLABELED`? That way devs can have icon-only tabs without the empty space reserved for the title.